packages/grafana-ui/src/components/uPlot/MicroPlot.tsx

```diff
@@ -223,25 +262,48 @@ function rangeToMinMax(timeRange: RawTimeRange) {
 
 export function getUPlotStuff(props: Props, range: any) {
   const { data, theme } = props;
-
   const series: uPlot.Series[] = [];
   const uData: any[] = [];
+
+  const dataFramesToPlot = data.filter(f => {
+    let { timeIndex } = getTimeField(f);
+    // filter out series without time index or if the time column is the only one (i.e. after transformations)
+    // won't live long as we gona move out from assuming x === time
+    return timeIndex !== undefined ? f.fields.length > 1 : false;
+  });
+
+  // uPlot data needs to be aligned on x-axis (ref. https://github.com/leeoniya/uPlot/issues/108)
+  // For experimentation just assuming alignment on time field, needs to change
+  const mergedDF = transformDataFrame(
+    [
+      {
+        id: 'seriesToColumns',
+        options: { byField: 'Time' },
+      },
+    ],
+    dataFramesToPlot
+  )[0];
```

Comment:
this logic should not be in the main ui component -- i think it belongs in the panels -- maybe we can add some utility functions that do the standard "merge everything"

Reply:
@ryantxu yes, this is me playing around, this won't belong to the graph itself. I plan to open some PR with my work around graph ltr this week.